core/src/utilities/helpers/navigation-helpers.js

```diff
@@ -0,0 +1,61 @@
+import { LuigiConfig } from '../../services/config';
+import { isLoggedIn } from './auth-helpers';
+export const isNodeAccessPermitted = (
+  nodeToCheckPermissionFor,
+  parentNode,
+  currentContext
+) => {
+  if (LuigiConfig.isAuthorizationEnabled() && !isLoggedIn()) return false;
+  const permissionCheckerFn = LuigiConfig.getConfigValue(
+    'navigation.nodeAccessibilityResolver'
+  );
+  if (typeof permissionCheckerFn !== 'function') {
+    return true;
+  }
+  return permissionCheckerFn(
+    nodeToCheckPermissionFor,
+    parentNode,
+    currentContext
+  );
+};
+
+export const applyContext = (context, addition, navigationContext) => {
+  if (addition) {
+    for (var p in addition) {
+      context[p] = addition[p];
+    }
+  }
+  if (navigationContext) {
+    context.parentNavigationContexts.unshift(navigationContext);
+  }
+  return context;
+};
+
+export const groupBy = (nodes, property) => {
```

Comment:
We could rename this fn to groupNodesByProperty  or groupNodesBy, since the function looks like you could just group anything.